symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py

```diff
@@ -38,27 +39,32 @@
 intermolecular_force_potential = Function("intermolecular_force_potential", units.energy, real=True)
 temperature = Symbol("temperature", units.temperature, positive=True)
 
+infinite_distance = Quantity(S.Infinity, dimension=units.length)
```

Comment:
Why do we even treat infinity different from zero, dimension-wise? Both of them consume whatever is multiplied with them: $`0 \cdot x = 0 \, \forall x \in \mathbb{C}`$ and $`\infty \cdot x = \infty \, \forall x \in \bar{\mathbb{C}} \setminus \{0\}`$. So technically infinity would be dimensionless the same way zero is.

Reply:
I'm not sure about it. 0 joules and 0 meters is the same value, because there is nothing to compare. But infinity is some huge amount of something, so inifinite amount of energy is not the same as infinite distance.
However in most physical applications infinity will be reduced to zero, and its dimension will likely will be erased anyway.

Reply:
Well, I've made inifinity to be compatible with any dimension, like zero. Not sure if it was a good idea, but code looks simpler, and dimensional analysis with infinity was broken anyway.

Reply:
I think it is good for now. For now it can probably be viewed as a limitation of sympy and our library.